tsdb/chunkenc/chunk_test.go

```diff
@@ -120,32 +119,27 @@ func benchmarkIterator(b *testing.B, newChunk func() Chunk) {
 				break
 			}
 			a.Append(p.t, p.v)
-			i++
 			j++
 		}
-		chunks = append(chunks, c)
 	}
 
 	b.ReportAllocs()
 	b.ResetTimer()
 
-	fmt.Println("num", b.N, "created chunks", len(chunks))
-
-	res := make([]float64, 0, 1024)
-
+	var res float64
 	var it Iterator
-	for i := 0; i < len(chunks); i++ {
-		c := chunks[i]
-		it := c.Iterator(it)
+	for i := 0; i < b.N; {
+		it := chunk.Iterator(it)
 
 		for it.Next() {
 			_, v := it.At()
-			res = append(res, v)
+			res = v
+			i++
```

Comment:
why do we even need this?

I think the only useful bench here is that passing an existing `it` to  `it := chunk.Iterator(it)` saves memory allocation.

Reply:
Which line(s) are you questioning?
 * `i++` is needed so we run the benchmark an appropriate number of times.
 * `res = v` is not needed, but I wanted to give some nod to the previous code, perhaps it was hoping to avoid the work being optimised out.

Reply:
wouldn' something like this do the same job?

```
	for i := 0; i < b.N; i++ {
```

Reply:
hm I tested it and it returns a completely different results.

Reply:
with your version the results are:

```
BenchmarkXORIterator-8           1000000              1075 ns/op               0 B/op          0 allocs/op
```

and with my syggestion:

```
BenchmarkXORIterator-8              5000            274289 ns/op              96 B/op          1 allocs/op
```

Reply:
Well, yes, mine is trying to do the same amount of work as before, just without gigabytes of memory in the set-up.
Yours is perfectly reasonable, but not comparable between versions.

Reply:
Strange, looking at the code:`i` only determines how many times to run the loop and it would stop executing only when few iterations produce similar/same results right? What am I missing?

```
it := chunk.Iterator(it) ---> it = chunk.Iterator(it)
```
to avoid allocating a new iterator.

Reply:
In the original code and my version, `b.N` determines the number of samples to read.
In your version, `b.N` chunks are read, and each chunk has 250 samples so it does 250x as much work per iteration.

Reply:
Makes sense to me.

@krasi-georgiev to answer your question, b.N will be limit for samples not chunks, just different unit - should be good enough.